File: spark-study-scala/src/main/java/com/zhouq/spark/sql/ManuallySpecifyOptions.scala
package com.zhouq.spark.sql

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.{DataFrame, SQLContext}

/**
  * Create by zhouq on 2019/9/9
  *
  */
object ManuallySpecifyOptions {
  def main(args: Array[String]): Unit = {
    val conf: SparkConf = new SparkConf().setAppName("ManuallySpecifyOptions").setMaster("local")
    val sc = new SparkContext(conf)

    val sqlContext = new SQLContext(sc)

    val usersDF: DataFrame = sqlContext.read
      .format("json")
      .load("H:\\bigdatatest\\spark\\sql\\people.json")

    usersDF.select("name").write
      .format("parquet")
      .save("H:\\bigdatatest\\spark\\sql\\out\\people_name_scala")

    sc.stop()
  }

}

File: spark-study-java/src/main/java/com/zhouq/spark/sql/ManuallySpecifyOptions.java
package com.zhouq.spark.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

/**
 * 手动指定数据源类型
 *
 * Create by zhouq on 2019/9/9
 *
 */
public class ManuallySpecifyOptions {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local").setAppName("ManuallySpecifyOptions");
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);

        DataFrame usersDF = sqlContext.read()
                .format("json")
                .load("H:\\bigdatatest\\spark\\sql\\people.json");

        usersDF.write()
                // 可选项 "parquet", "json"
                .format("json")
                .save("H:\\bigdatatest\\spark\\sql\\out\\people2");

        sc.stop();
    }
}
